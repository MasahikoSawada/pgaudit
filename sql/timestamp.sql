-- Create test table
CREATE TABLE timestamp_test1(col timestamp);
CREATE TABLE timestamp_test2(col timestamp);

-- Check if the audit logging for timestamp_test1 is logged and
-- the audit logging for timestamp_test2 is NOT logged now.
SELECT * FROM timestamp_test1;
SELECT * FROM timestamp_test2;

-- Wait until past the timestamp range.
SELECT pg_sleep(extract(second from (date_trunc('min', now() + '1 min')  - now()))::int + 3);

-- Check if the audit logging for timestamp_test1 is NOT logged and
-- the audit logging for timestamp_test2 is logged now.
SELECT * FROM timestamp_test1;
SELECT * FROM timestamp_test2;




