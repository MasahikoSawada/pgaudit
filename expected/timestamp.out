-- Create test table
CREATE TABLE timestamp_test1(col timestamp);
CREATE TABLE timestamp_test2(col timestamp);
-- Check if the audit logging for timestamp_test1 is logged and
-- the audit logging for timestamp_test2 is NOT logged now.
SELECT * FROM timestamp_test1;
NOTICE:  AUDIT: SESSION,1,1,READ,,regression,[local],pg_regress,,,,,SELECT,TABLE,public.timestamp_test1,SELECT * FROM timestamp_test1;,<none>
 col 
-----
(0 rows)

SELECT * FROM timestamp_test2;
 col 
-----
(0 rows)

-- Wait until past the timestamp range.
SELECT pg_sleep(extract(second from (date_trunc('min', now() + '1 min')  - now()))::int + 3);
 pg_sleep 
----------
 
(1 row)

-- Check if the audit logging for timestamp_test1 is NOT logged and
-- the audit logging for timestamp_test2 is logged now.
SELECT * FROM timestamp_test1;
 col 
-----
(0 rows)

SELECT * FROM timestamp_test2;
NOTICE:  AUDIT: SESSION,2,1,READ,,regression,[local],pg_regress,,,,,SELECT,TABLE,public.timestamp_test2,SELECT * FROM timestamp_test2;,<none>
 col 
-----
(0 rows)

